plugin/evm/orderbook/contract_events_processor.go

```diff
@@ -215,6 +228,63 @@ func (cep *ContractEventsProcessor) handleOrderBookEvent(event *types.Log) {
 	}
 }
 
+func (cep *ContractEventsProcessor) handleIOCOrderBookEvent(event *types.Log) {
+	removed := event.Removed
+	args := map[string]interface{}{}
+	switch event.Topics[0] {
+	case cep.iocOrderBookABI.Events["OrderPlaced"].ID:
+		err := cep.iocOrderBookABI.UnpackIntoMap(args, "OrderPlaced", event.Data)
+		if err != nil {
+			log.Error("error in iocOrderBookABI.UnpackIntoMap", "method", "OrderPlaced", "err", err)
+			return
+		}
+		orderId := event.Topics[2]
+		if !removed {
+			order := IOCOrder{}
+			order.DecodeFromRawOrder(args["order"])
+			limitOrder := Order{
```

Comment:
You also need to write orderType and expireAt no?

Reply:
`*IOCOrder.DecodeFromRawOrder()` takes care of that